service/matching/task_queue_partition_manager.go

```diff
@@ -69,13 +70,15 @@ type (
 		// is delegated to the defaultQueue.
 		defaultQueue physicalTaskQueueManager
 		// used for non-sticky versioned queues (one for each version)
-		versionedQueues     map[string]physicalTaskQueueManager
-		versionedQueuesLock sync.RWMutex // locks mutation of versionedQueues
-		userDataManager     userDataManager
-		logger              log.Logger
-		throttledLogger     log.ThrottledLogger
-		matchingClient      matchingservice.MatchingServiceClient
-		metricsHandler      metrics.Handler // namespace/taskqueue tagged metric scope
+		versionedQueues             map[string]physicalTaskQueueManager
+		versionedQueuesLock         sync.RWMutex // locks mutation of versionedQueues
+		userDataManager             userDataManager
+		logger                      log.Logger
+		throttledLogger             log.ThrottledLogger
+		matchingClient              matchingservice.MatchingServiceClient
+		metricsHandler              metrics.Handler                                                          // namespace/taskqueue tagged metric scope
+		cachedPhysicalInfoByBuildId map[string]map[enumspb.TaskQueueType]*taskqueuespb.PhysicalTaskQueueInfo // non-nil for root-partition
+		lastFanOut                  atomic.Int64                                                             // serves as a TTL for cachedPhysicalInfoByBuildId
```

Comment:
consider calling this `lastFanOutTime`?

Reply:
I don't particularly mind doing that but the convention I followed was taken from the way we track poll time inside of matcher. https://github.com/temporalio/temporal/blob/898c4b8ce8f1169c54e633fa212c2549db1ec483/service/matching/matcher.go#L75

Do we wanna keep as this is for consistency purposes? I don't have a strong preference for either.